Client/mods/deathmatch/logic/CClientVectorGraphic.cpp

```diff
@@ -18,7 +18,7 @@ CClientVectorGraphic::CClientVectorGraphic(CClientManager* pManager, ElementID I
 
     m_pManager = pManager;
 
-    m_pVectorGraphicDisplay = std::make_unique<CClientVectorGraphicDisplay>(m_pManager->GetDisplayManager(), this);
```

Comment:
Why wasn't std::shared_ptr used here?

Reply:
Because we want to ensure the manager handles cleanup/destruction of display instances.

And actually for that reason, this should simply be a reference. We'll have the display manager actually create the `CClientVectorGraphicDisplay` instance, e.g: `auto& display = m_pManager->GetDisplayManager()->CreateVectorDisplay(this);`

It makes more sense in terms of the manager/mediator usage, and we guarantee the lifetime/ownership without additional overhead of a `shared_ptr` which seems slightly overkill for the purpose.

Reply:
> Because we want to ensure the manager handles cleanup/destruction of display instances.

But you never destruct  m_pVectorGraphicDisplay.

Reply:
Yes, you're right - I'll change it as mentioned which will resolve this issue

Reply:
@Lpsd Any updates on that?

> Yes, you're right - I'll change it as mentioned which will resolve this issue

Reply:
@TheNormalnij your approach using shared_ptr was actually easier due to spaghetti 'pattern' used for this display system. Thanks!